x/ethbridge/types/msgs.go

```diff
@@ -152,6 +168,76 @@ func (msg MsgBurn) GetSigners() []sdk.AccAddress {
 	return []sdk.AccAddress{cosmosSender}
 }
 
+// GetProphecyID get prophecy ID for lock message
+func (msg MsgBurn) GetProphecyID(doublePeggy bool, sequence, globalNonce uint64) []byte {
+
+	return ComputeProphecyID(
+		msg.CosmosSender,
+		sequence,
+		msg.EthereumReceiver,
+		// TODO need get the token address from token's symbol
+		msg.EthereumReceiver,
+		msg.Amount,
+		doublePeggy,
+		globalNonce,
+		msg.NetworkDescriptor,
+	)
+}
+
+// ComputeProphecyID compute the prophecy id
+func ComputeProphecyID(cosmosSender string, sequence uint64, ethereumReceiver string, tokenAddress string, amount sdk.Int,
+	doublePeggy bool, globalNonce uint64, networkDescriptor oracletypes.NetworkDescriptor) []byte {
+
+	bytesTy, _ := abi.NewType("bytes", nil)
+	boolTy, _ := abi.NewType("bool", nil)
+	uint128Ty, _ := abi.NewType("uint128", nil)
+	uint256Ty, _ := abi.NewType("uint256", nil)
+	addressTy, _ := abi.NewType("address", nil)
+
+	arguments := abi.Arguments{
```

Comment:
@daniportugalgit Please take a look at this and double check that everything is in order per the new smart contract data structures.

Reply:
Although the ClaimData does have new fields, the ProphecyID doesn't.
This structure is correct.